modules/gapi/src/streaming/vpl/vpl_dx11_accel.cpp

```diff
@@ -0,0 +1,171 @@
+#ifdef HAVE_ONEVPL
+#include "streaming/vpl/vpl_dx11_accel.hpp"
+#include "streaming/vpl/vpl_utils.hpp"
+#include "streaming/vpl/surface/frame_adapter.hpp"
+#include "streaming/vpl/surface/surface.hpp"
+#include "logger.hpp"
+
+#ifdef HAVE_DIRECTX
+    #ifdef HAVE_D3D11
+        #pragma comment(lib,"d3d11.lib")
+        //#pragma comment(lib,"d3d10.lib")
+
+        #define D3D11_NO_HELPERS
+        #include <d3d11.h>
+        #include <d3d11_4.h>
+        #include <codecvt>
+        #include "opencv2/core/directx.hpp"
+        #ifdef HAVE_OPENCL
+            #include <CL/cl_d3d11.h>
+        #endif
+
+namespace cv {
+namespace gapi {
+namespace wip {
+VPLDX11AccelerationPolicy::VPLDX11AccelerationPolicy()
+{
+#ifdef CPU_ACCEL_ADAPTER
+    adapter.reset(new VPLCPUAccelerationPolicy);
+#endif
+}
+
+VPLDX11AccelerationPolicy::~VPLDX11AccelerationPolicy()
+{
+    if (hw_handle)
+    {
+        GAPI_LOG_INFO(nullptr, "VPLDX11AccelerationPolicy release ID3D11Device");
+        hw_handle->Release();
+    }
+}
+
+void VPLDX11AccelerationPolicy::init(session_t session) {
+#if 0 /* Activate it for LEGACY API*/
```

Comment:
?

Reply:
will activate it in next PR due to DX11 enabling